### PR TITLE
fix: recover password flow in identifier-first mode

### PR DESCRIPTION
**Pull request by aarongranick-okta**
## Description:
password recovery flow has extra steps in identifier-first flow. In order to keep the user experience consistent with identify-recovery flow, the `okta_password` authenticator should be specified after the `identify` remediation.

- shows `identify-recovery` UI instead of `identify` when running with flow = `resetPassword`
- selects `okta_password` as the authenticator on submission of the `identify` remediation
- also fixes autocomplete on identify recovery view

## PR Checklist

- [ ] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Issue:

- [OKTA-462165](https://oktainc.atlassian.net/browse/OKTA-462165)

### CODE REVIEW

File: playground/mocks/config/responseConfig.js [hunk 1]
```diff
@@ -1047,5 +1047,5 @@ const selectOktaVerifyMethod = {
 };
 
 module.exports = {
-  mocks: idx
+  mocks: Test.EnrollAuthenticatorWebAuthn.mock
```
Comment (jaredperreault-okta) [inline playground/mocks/config/responseConfig.js:1050]: Is this on purpose?

File: yarn.lock [hunk 2]
```diff
@@ -2995,27 +2987,6 @@
...
-    cross-fetch "^3.0.6"
-    js-cookie "2.2.1"
-    node-cache "^5.1.2"
-    p-cancelable "^2.0.0"
-    text-encoding "^0.7.0"
-    tiny-emitter "1.1.0"
-    webcrypto-shim "^0.1.5"
-    xhr2 "0.1.3"
-
 "@okta/okta-auth-js@^6.4.0":
```
Comment (shuowu-okta) [inline yarn.lock:2990]: maybe run add/upgrade again to stay with the latest patch.

**Review by shuowu-okta — APPROVED** [review]

File: yarn.lock (hunk 2 above)
Comment (aarongranick-okta) [reply yarn.lock:2990]: I'm running tests with beta version, will run again after promotion of auth-js w/ needed patches.

File: test/testcafe/spec/Identify_spec.js [hunk 3]
```diff
@@ -125,8 +125,8 @@ test.requestHooks(identifyRequestLogger, identifyMock)('should be able to submit
   await t.expect(req.method).eql('post');
   await t.expect(req.url).eql('http://localhost:3000/idp/idx/identify');
   await t.expect(reqHeaders['x-device-fingerprint']).notOk(); // only enabled if `features.deviceFingerprinting` is true
-  const authJSVersion = getAuthJSVersion();
-  await t.expect(reqHeaders['x-okta-user-agent-extended']).eql(`okta-auth-js/${authJSVersion} okta-signin-widget-${config.version}`);
+  // const authJSVersion = getAuthJSVersion();
+  // await t.expect(reqHeaders['x-okta-user-agent-extended']).eql(`okta-auth-js/${authJSVersion} okta-signin-widget-${config.version}`);
```
Comment (jaredperreault-okta) [inline test/testcafe/spec/Identify_spec.js:129]: remove?

**Review by jaredperreault-okta — APPROVED** [review]

File: playground/mocks/config/responseConfig.js (hunk 1 above)
Comment (aarongranick-okta) [reply playground/mocks/config/responseConfig.js:1050]: will be removed before merge. I need to promote auth-js and update/test again. I'll clean this up too

**Pull request closed without merging**